### DISCUSSION TITLE
Public URL help

### DISCUSSION THREAD

**Discussion by ae8u** [alandtse/alexa_media_player#2664] [Q&A]
I am trying to set up and use Nabu Casa tts (or Amazon Polly tts) with Alexa Media Player. But it tells me I need to enter my public url. When I do so, I am entering it in the configuration.yaml as "https://xxxx.link:8443". Please note that I use 8443 rather than 8123 for a little extra security, and it has been working fine for several years. But when I restart Home Assistant I get the following error:

2024-11-02 10:35:31.096 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration alexa_media which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2024-11-02 10:35:42.418 DEBUG (MainThread) [custom_components.alexa_media] Importing config information for m****t@a******t - https://xxxx.link:8443 from configuration.yaml
2024-11-02 10:35:42.418 DEBUG (MainThread) [custom_components.alexa_media] Found existing config entries
2024-11-02 10:35:42.418 DEBUG (MainThread) [custom_components.alexa_media] Creating new config entry
2024-11-02 10:35:42.418 ERROR (MainThread) [homeassistant.setup] Error during setup of component alexa_media
File "/config/custom_components/alexa_media/__init__.py", line 189, in async_setup

Can anyone provide any help to solve this?

**Comment by danielbrunt57** [comment]
YAML setup is going to be deprecated when someone gets around to doing it therefore, you really should migrate to the newer config entry method for setting up the integration.

I suspect the issue lies somewhere in the legacy YAML processing logic in that it's likely missing the `public_url` somewhere which got missed when `public_url` became a reconfigurable entry in the newer config entry method.

The logic should test the for YAML `email` & `password` and if found execute the section which logs "Updating existing entry" which is does not have `public_url`:

```
                if (
                    entry.data.get(CONF_EMAIL) == account[CONF_EMAIL]
                    and entry.data.get(CONF_URL) == account[CONF_URL]
                ):
                    _LOGGER.debug("Updating existing entry")
                    hass.config_entries.async_update_entry(
                        entry,
                        data={
                            CONF_EMAIL: account[CONF_EMAIL],
                            CONF_PASSWORD: account[CONF_PASSWORD],
                            CONF_URL: account[CONF_URL],
                            CONF_INCLUDE_DEVICES: account[CONF_INCLUDE_DEVICES],
                            CONF_EXCLUDE_DEVICES: account[CONF_EXCLUDE_DEVICES],
                            CONF_SCAN_INTERVAL: account[
                                CONF_SCAN_INTERVAL
                            ].total_seconds(),
                            CONF_QUEUE_DELAY: account[CONF_QUEUE_DELAY],
                            CONF_OAUTH: account.get(
                                CONF_OAUTH, entry.data.get(CONF_OAUTH, {})
                            ),
                            CONF_OTPSECRET: account.get(
                                CONF_OTPSECRET, entry.data.get(CONF_OTPSECRET, "")
                            ),
                            CONF_EXTENDED_ENTITY_DISCOVERY: account[
                                CONF_EXTENDED_ENTITY_DISCOVERY
                            ],
                            CONF_DEBUG: account[CONF_DEBUG],
                        },
                    )
                    entry_found = True
                    break
```
But yours is executing the section which logs "Creating new config entry" which references `public_url`:
```
        if not entry_found:
            _LOGGER.debug("Creating new config entry")
            hass.async_create_task(
                hass.config_entries.flow.async_init(
                    DOMAIN,
                    context={"source": SOURCE_IMPORT},
                    data={
                        CONF_URL: account[CONF_URL],
                        CONF_EMAIL: account[CONF_EMAIL],
                        CONF_PASSWORD: account[CONF_PASSWORD],
                        CONF_PUBLIC_URL: account[CONF_PUBLIC_URL],
                        CONF_INCLUDE_DEVICES: account[CONF_INCLUDE_DEVICES],
                        CONF_EXCLUDE_DEVICES: account[CONF_EXCLUDE_DEVICES],
                        CONF_SCAN_INTERVAL: account[CONF_SCAN_INTERVAL].total_seconds(),
                        CONF_QUEUE_DELAY: account[CONF_QUEUE_DELAY],
                        CONF_OAUTH: account.get(CONF_OAUTH, {}),
                        CONF_OTPSECRET: account.get(CONF_OTPSECRET, ""),
                        CONF_EXTENDED_ENTITY_DISCOVERY: account[
                            CONF_EXTENDED_ENTITY_DISCOVERY
                        ],
                        CONF_DEBUG: account[CONF_DEBUG],
                    },
                )
            )
    return True
```
I'm not sure what the correct/best fix is.

I think your best bet to quickly resolve your issue is to remove your yaml entries, restart HA and add the integration via Settings > Integrations > Add Integration which is the config entry method.
